src/pages/exchange-crypto/exchange-crypto.ts

```diff
@@ -66,217 +103,508 @@ export class ExchangeCryptoPage {
+    failedPromises.forEach(promise => {
+      switch (promise.exchange) {
+        case '1inch':
+          this.logger.error('1Inch getCurrencies Error');
+          if (promise.reason && promise.reason.message)
+            this.logger.error(promise.reason.message);
+          break;
+
+        case 'changelly':
+          this.logger.error('Changelly getCurrencies Error');
+          if (promise.reason && promise.reason.message)
+            this.logger.error(promise.reason.message);
+          break;
+
+        default:
+          break;
+      }
+    });
+
+    successfulPromises.forEach(promise => {
+      switch (promise.exchange) {
+        case '1inch':
+          if (!promise.data || !promise.data.tokens) {
+            this.logger.error('1Inch getCurrencies Error');
+            // this.showErrorAndBack(
+            //   null,
+            //   this.translate.instant(
+            //     '1Inch is not available at this moment. Please, try again later.'
+            //   )
+            // );
+            return;
+          }
+
+          this.oneInchSupportedCoins;
+          this.oneInchAllSupportedCoins = [];
+          this.oneInchAllSupportedCoinsSymbols = [];
+
+          _.forEach(Object.keys(promise.data.tokens), key => {
+            this.oneInchAllSupportedCoins.push(promise.data.tokens[key]);
+            this.oneInchAllSupportedCoinsSymbols.push(
+              promise.data.tokens[key].symbol.toLowerCase()
+            );
+          });
+
+          if (
+            _.isArray(this.oneInchAllSupportedCoins) &&
+            this.oneInchAllSupportedCoins.length > 0
+          ) {
+            this.oneInchSupportedCoins = _.intersection(
+              this.currencyProvider.getAvailableCoins(),
+              this.oneInchAllSupportedCoinsSymbols
+            );
+          }
+
+          this.oneInchSupportedCoinsFull = this.oneInchAllSupportedCoins.filter(
+            token => {
+              return this.oneInchSupportedCoins.includes(
+                token.symbol.toLowerCase()
               );
-              this.supportedCoins.splice(index, 1);
             }
-          });
-        }
+          );
 
-        this.logger.debug('Changelly supportedCoins: ' + this.supportedCoins);
+          this.logger.debug(
+            '1Inch supportedCoins: ' + this.oneInchSupportedCoins
+          );
+          break;
 
-        this.allWallets = this.profileProvider.getWallets({
-          network: 'livenet',
-          onlyComplete: true,
-          coin: this.supportedCoins,
-          backedUp: true
-        });
+        case 'changelly':
+          if (promise.data.error) {
+            this.logger.error(
+              'Changelly getCurrencies Error: ' + promise.data.error.message
+            );
+            // this.showErrorAndBack(
+            //   null,
+            //   this.translate.instant(
+            //     'Changelly is not available at this moment. Please, try again later.'
+            //   )
+            // );
```

Comment:
and this one

Reply:
This error handle is no longer necessary. I removed it